source/admin_manual/migrating_mailboxes.rst

```diff
@@ -0,0 +1,308 @@
+    You can do this by logging in using your old POP3 server, issuing UIDL command and saving the output.
+    Then log in using Dovecot and save its UIDL output as well. Use e.g. diff command to verify that the lists are identical. Note that:
+
+     *  If a client already saw changed UIDLs and decided to start re-downloading mails, it's unlikely there is anything you can do to stop it. Even going back to your old server is unlikely to help at that point.
+     *  Some (many?) POP3 clients also require that the message ordering is preserved.
+     *  Some clients re-download all mails if you change the hostname in the client configuration. Be aware of this when testing.
+
+
+Migration from Gmail to Dovecot
+===============================
+
+You can use dsync migration via IMAP protocol, but there are a few things different with Gmail compared to other IMAP servers.
+With Gmail when you delete a mail from POP3, the mail is only hidden from future POP3 sessions, but it's still available via IMAP. If you wish to preserve this functionality, there's a pop3_deleted_flag setting in Dovecot v2.2.2+.
+
+Gmail has labels. If a message has multiple labels, it shows up in multiple IMAP folders, but it's still the same message and uses quota only once for that message.
+Dovecot currently doesn't have such support, so the migration will copy the message to multiple folders and each instance will use up quota.
+There's currently no easy fix for this, although there are some future plans to optionally not count message copies towards quota.
+
+Even though the quota is duplicated, it doesn't mean that the storage usage has to be duplicated. Use the dsync -v parameter to avoid this.
+See https://github.com/dovecot/core/commit/70df8f39fb3db7c49b18c855178f8172176a037a
+
+Gmail has virtual folders: "All Mail", "Starred" and "Important". From migration point of view this means that the migration should skip most of these folders,
+since their mails are in other folders anyway.
+With v2.2.3+ you can tell dsync to skip these folders::
+
+  doveadm sync -x '\Flagged' -x '\Important'
+
+by using the ``\flag`` parameters dsync finds the folders by their ``SPECIAL-USE`` flag rather than their name (which may be different for different user depending on their language).
+
+The "All Mail" folder actually contains also "archived mails" that don't exist in any other folder. These mails need to be migrated. See below.
+
+Google requires that SSL/TLS be enabled to connect through IMAP. Make sure that the following are enabled in your Dovecot configuration and set to appropriate values based on your distribution (usually either one is enough):
+
+    * ssl_client_ca_dir = (''your distribution's trusted TLS CA store (Fedora / CentOS / Redhat uses /etc/pki/tls/ ))''
+    * ssl_client_ca_file = (''your distribution's trusted TLS CA file (Fedora / CentOS / Redhat uses /etc/pki/tls/cert.pem ))''
```

Comment:
I think this SSL stuff can be removed, since it's duplication of the earlier stuff in this page.